apple/swift-docc
# [ConvertService] Support resolving images referenced in content

Bug/issue #, if applicable: rdar://107531364

## Summary

This enables convert service requests to resolve assets that are references in the content. This required two changes:
- Allow the fallback asset resolve to look for an asset identifier when the local asset manager didn't find one.
- When rendering the image, get the file extension from the asset's url instead of from the asset identifier.

This PR also removes unnecessary empty media references when an asset didn't render anything.

## Dependencies

None

## Testing

In a client that uses` ConvertService`:
- Verify that passing an article with image references (either using `![]()`, `@Image`, or `@PageImage`) result in DocC returning a render node with image references for those images. 
- Add another article with `@PageImage` images and link it with `@Links(visualStyle: detailedGrid)` and verify that DocC returns a image references for the second article's page images.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
